docs/making-accessible-react-native-components.md

```diff
@@ -0,0 +1,252 @@
+
+| API                                     | Version of React Native | Version of @types/react-native | Removed version |
+| --------------------------------------- | ----------------------- | ------------------------------ | --------------- |
+| accessible                              | 0.5                     | first release                  | N/A             |
+| ---                                     | ---                     | ---                            | ---             |
+| accessibilityLabel                      | 0.5                     | first release                  | N/A             |
+| accessibilityRole                       | 0.57                    | 0.57                           | N/A             |
+| accessibilityState                      | 0.61                    | 0.6                            | N/A             |
+| accessbilityValue                       | 0.62                    | 0.62                           | N/A             |
+| accessibilityActions                    | 0.6                     | 0.6                            | N/A             |
+| ~~accessibilityTrait (iOS)~~            | 0.5                     | first release                  | 0.6             |
+| ~~accessbilityComponentType (Android)~~ | 0.5                     | first release                  | 0.6             |
+| ~~accessibilityStates~~                 | 0.57                    | 0.57                           | 0.62            |
+
+## Notes Based on Components
+
+### Button/Touchable
+
+1. Set accessibility label.
+2. Set accessibility role such as button, imagebutton or link on the Touchables.
+3. Set accessibility state properly such as "disabled" if the component can be disabled.
+
+### Switch/Radio Button/Checkbox
+
+1. Ideally the switch/radio button/checkbox should be grouped with the adjacent text as a single selectable component. First, set`accessible={true}`on the container which wraps the text and the switch/radio button. Second, set `importantForAccessibility='no-hide-descendants'` on the switch/radio button/checkbox for the sake of Android.
+
+> On Android, {accessible} doesn't work as described in the API document if some elements in the group are accessible. Setting this prop will result in multiple accessibility focus - one on the group view and the other on the accessible child elements. Therefore, we need to hide the accessible elements using importForAccessibility API. This API is android specific but it does no harm on iOS.
+
+1. Set accessibility label. Do not include the state of the component such as "on/off" or "checked/unchecked" in the label.
+2. Set accessibility role.
+3. Set accessibility state "checked" properly. If the component can be disabled, set "disabled" state as well.
+4. If the switch and the text are wrapped in a container, add accessibility action "activate" on the container. Toggle the switch/radio button in the onAccessibilityAction function.
+
+### _Sample Code_
```

Comment:
Can you convert this code block to the SnackPlayer ([example](https://github.com/facebook/react-native-website/blame/master/docs/activityindicator.md#L10))? It would be nice to offer both syntax version (functional and class), but if only one will be used it should be based on the functional approach.

Reply:
Hi @Simek, the alex check gave warnings for the following item:
docs/making-accessible-react-native-components.md
  7:124-7:130  warning  `easily` may be insensitive, try not to use it  easy  retext-equality

Do you have any suggestion about which word to use instead of 'easily'?

Reply:
As for converting the code block to the SnackPlayer, the sample code I provide is code snippet. It's not complete. I just want to show how to appropriately set the accessibility properties. It doesn't aim to demo a UI component. Also, the accessibility features cannot be tested in the SnackPlayer. I don't see the value to show an plain view in the SnackPlayer. Please let me know what you think. Thanks.

Reply:
> Also, the accessibility features cannot be tested in the SnackPlayer. 

I'm still leaning to rewriting/completing the examples and converting them to Snacks. There might be some feature unavailable in the preview, but every Snack can also be run on the user device (where accessibility code should work correctly).

It would be nice if you can also switch from classes to the functional approach (or add same example in both syntaxes and utilize tabs). Currently on the website we are promoting functions over classes.

Reply:
Sorry for the late respond. I'm a bit tied up at the moment. I'll look into converting the example when I have time.